src/OrbitGl/TrackManager.cpp

```diff
@@ -73,7 +77,7 @@ std::vector<Track*> TrackManager::GetAllTracks() const {
 }
 
 std::vector<FrameTrack*> TrackManager::GetFrameTracks() const {
-  std::lock_guard<std::recursive_mutex> lock(mutex_);
+  absl::MutexLock lock(&mutex_);
```

Comment:
Probably not related to this PR but maybe worth considering: 

I saw this pattern a few times in this class.

- Get a lock
- Copy guts from shared pointers into raw pointers and ...
- return the raw pointers while releasing the lock.

This might be fine but looks dangerous - subsequent modifications to the shared pointer will turn the exported raw pointers into rubbish.

Anyway - that is probaly not something to fix in this PR.

Reply:
Yes, I'm aware of this.

My plan was to directly get rid of the multi-threading in this class and directly create the Tracks before rendering, moving the mutex to a TrackDataManager class.

See the proposed intermediate step in http://go/stadia-orbit-track-creation#heading=h.vpgt3chmpmpb. It wasn't a priority though.